src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java

```diff
@@ -60,97 +60,63 @@ public void setHttpRequest(HttpRequest request) {
     }
 
     /**
-     * This method only extracts the event information from ER2.0 based on the
-     * eventID.
+     * This method is used to fetch only the upstream or downstream or both event information from
+     * ER based on the eventID and searchOption conditions.
      *
-     * @param eventId
-     *            the id of the event.
+     * @param eventId      the id of the event.
+     * @param searchOption the SearchOption to indicate whether to search up, down or both ways from
+     *                     the eventId.
+     * @param limit        sets the limit of how many events up and/or down stream from the eventId
+     *                     to include in the result.
+     * @param levels       sets the limit of how many levels up and/or down stream from the eventId
+     *                     to include in the result.
+     * @param tree         whether or not to retain the tree structure in the result.
      * @return ResponseEntity
+     * @throws PropertyNotFoundException 
      */
-    public ResponseEntity getEventDataById(String eventId) {
-        String erUrl = null;
+    public ResponseEntity getEventStreamDataById(String eventId, SearchOption searchOption,
+            int limit,
+            int levels, boolean tree) throws PropertyNotFoundException, Exception {
 
+        String uri = null;
```

Comment:
This parameter does not help very much after the refactoring. Maybe we should remove it

Reply:
Ah yes, it's always null, I've removed it.